### DISCUSSION TITLE
ApplyLayer stdout:  stderr: remount /, flags: 0x44000: permission denied exit status 1" Error: creating build container:

### DISCUSSION THREAD

**Discussion by shay1197** [containers/podman#24310] [Q&A]
HI,
I am having error running in a rootless container (ubuntu 22.04) in kubernetes ( podman build)
+ podman build --storage-driver=vfs --no-cache -t blabla.bla:48444/jenkins-agent:3336-jdk17 --tls-verify=false .
[1/12] STEP 1/13: FROM jenkins/inbound-agent:3256.v88a_f6e922152-4-jdk17 AS base
Resolving "jenkins/inbound-agent" using unqualified-search registries (/etc/containers/registries.conf)
Trying to pull docker.io/jenkins/inbound-agent:3256.v88a_f6e922152-4-jdk17...
Getting image source signatures
Copying blob sha256:c10e49e2ae751d7e0a95fd985d4baf869ddf847c9dff496af287618d9f4a402d
Copying blob sha256:d3915ffa03dd97aeff52883276bf9048d5f45fee6f5c274df6877acac5d35abb
Copying blob sha256:a445b84e64466cf2b1855021971fe4554a302b17009a8df7ae676501e4bb59e0
Copying blob sha256:55867a3b2a3b8dad7b0d2be7659ca4b6e9c6785587ec479ab8b39da364b706d4
Copying blob sha256:f11c1adaa26e078479ccdd45312ea3b88476441b91be0ec898a7e07bfd05badc
Copying blob sha256:142683d3f5578ae8156707f6c20be3ceb67e1b9f2ae3bb725b9448c193c38dae
Copying blob sha256:ab94afba5451bb12f89c75da5ccc33b3ad9238fe9646651f4ae2a412564dba07
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
Copying blob sha256:7af6470caf99a63d368b81ca0645a1ea3379ce4e40831453b81967c086dd4842
Copying blob sha256:2828fce91c7c2fcc8e3a77d4186e31468e20efac6ddbd287ab67576b6fc5df70
time="2024-10-17T23:50:23Z" level=error msg="While applying layer: ApplyLayer stdout:  stderr: remount /, flags: 0x44000: permission denied exit status 1"
Error: creating build container: copying system image from manifest list: writing blob: adding layer with blob "sha256:f11c1adaa26e078479ccdd45312ea3b88476441b91be0ec898a7e07bfd05badc"/""/"sha256:32148f9f6c5aadfa167ee7b146b9703c59307049d68b090c19db019fd15c5406": ApplyLayer stdout:  stderr: remount /, flags: 0x44000: permission denied exit status 1

adding podman info-
+ podman info
host:
  arch: amd64
  buildahVersion: 1.37.4
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - rdma
  - misc
  cgroupManager: cgroupfs
  cgroupVersion: v2
  conmon:
    package: conmon_2.1.12-1_amd64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.12, commit: unknown'
  cpuUtilization:
    idlePercent: 96.97
    systemPercent: 0.84
    userPercent: 2.19
  cpus: 4
  databaseBackend: sqlite
  distribution:
    codename: trixie
    distribution: debian
    version: unknown
  eventLogger: file
  freeLocks: 2048
  hostname: jenkins-agent-mqxt1
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.15.0-91-generic
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 363827200
  memTotal: 16766251008
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: Unknown
    package: netavark_1.4.0-3_amd64
    path: /usr/lib/podman/netavark
    version: netavark 1.4.0
  ociRuntime:
    name: crun
    package: crun_1.17-1_amd64
    path: /usr/bin/crun
    version: |-
      crun version 1.17
      commit: 000fa0d4eeed8938301f3bcf8206405315bc1017
      rundir: /tmp/storage-run-1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: ""
    package: ""
    version: ""
  remoteSocket:
    exists: false
    path: /tmp/storage-run-1000/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns_1.2.0-1_amd64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.4
  swapFree: 0
  swapTotal: 0
  uptime: 42h 9m 42.00s (Approximately 1.75 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /home/jenkins/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: vfs
  graphOptions: {}
  graphRoot: /home/jenkins/.local/share/containers/storage
  graphRootAllocated: 98743734272
  graphRootUsed: 58775232512
  graphStatus: {}
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 0
  runRoot: /tmp/storage-run-1000/containers
  transientStore: false
  volumePath: /home/jenkins/.local/share/containers/storage/volumes
version:
  APIVersion: 5.2.4
  Built: 1729158387
  BuiltTime: Thu Oct 17 09:46:27 2024
  GitCommit: ""
  GoVersion: go1.23.2
  Os: linux
  OsArch: linux/amd64
  Version: 5.2.4

Thanks  in advanced

**Comment by shahgnp** [comment]
I'm experiencing the same issue while trying to build a Docker image with buildah in a privileged GitLab Kubernetes executor.
```
$ buildah images
REPOSITORY   TAG   IMAGE ID   CREATED   SIZE
$ buildah bud --format "$BUILDAH_FORMAT" --storage-driver "$STORAGE_DRIVER" --isolation=chroot -t "$FQ_IMAGE_NAME" .
STEP 1/11: FROM python:3.9.2
Resolved "python" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull docker.io/library/python:3.9.2...
Getting image source signatures
Copying blob sha256:004f1eed87df3f75f5e2a1a649fa7edd7f713d1300532fd0909bb39cd48437d7
Copying blob sha256:5d6f1e8117dbb1c6a57603cb4f321a861a08105a81bcc6b01b0ec2b78c8523a5
Copying blob sha256:48c2faf66abec3dce9f54d6722ff592fce6dd4fb58a0d0b72282936c6598a3b3
Copying blob sha256:234b70d0479d7f16d7ee8d04e4ffdacc57d7d14313faf59d332f18b2e9418743
Copying blob sha256:6fa07a00e2f029c4b2c7f177a2b696f1b3510040cde4f5bb06ddbca98e7fbf76
Copying blob sha256:04a31b4508b8e95fb3cb25486c4068185054895b12e0611e386a002ee9c0e07c
Copying blob sha256:1823d93d969893b61303752fa11b710c4befae6f15a775fad2c2c0bcb619cf2e
Copying blob sha256:2eee1b8e253830859988454f07e1e98387cf5e3730e7eedee333aa7fd1a04be2
Copying blob sha256:b48672c0140e5d73a8b044e4c1c229f3cc88fd8ff645c7fc6f774c830ae1f8d4
time="2025-09-04T12:32:06Z" level=error msg="While applying layer: ApplyLayer stdout:  stderr: remount /, flags: 0x44000: permission denied exit status 1"
Error: creating build container: unable to copy from source docker://python:3.9.2: copying system image from manifest list: writing blob: adding layer with blob "sha256:004f1eed87df3f75f5e2a1a649fa7edd7f713d1300532fd0909bb39cd48437d7"/""/"sha256:4ef4adca5c3b96107661045d947bf51f1f27f47c2557c9eea367c32a8bc8c399": ApplyLayer stdout:  stderr: remount /, flags: 0x44000: permission denied exit status 1
```
These are my helm values.

```
gitlabUrl: https://gitlab.com
runnerRegistrationToken: xxx-my-token-xxx

unregisterRunners: true

## For RBAC support:
rbac:
  create: true
  rules:
    - resources: ["events"]
      verbs: ["list", "watch"]
    - resources: ["namespaces"]
      verbs: ["create", "delete"]
    - resources: ["pods"]
      verbs: ["create","delete","get","list","watch"]
    - apiGroups: [""]
      resources: ["pods/attach","pods/exec"]
      verbs: ["get","create","patch","delete"]
    - apiGroups: [""]
      resources: ["pods/log"]
      verbs: ["get","list"]
    - resources: ["secrets"]
      verbs: ["create","delete","get","update"]
    - resources: ["serviceaccounts"]
      verbs: ["get"]
    - resources: ["services"]
      verbs: ["create","get"]

  clusterWideAccess: false

  ## Use podSecurity Policy
  ## ref: https://kubernetes.io/docs/concepts/policy/pod-security-policy/
  podSecurityPolicy:
    enabled: true
    resourceNames:
      - gitlab-runner

serviceAccount:
  create: true
  name: "gitlab-runner"

runners:
  config: |
    [[runners]]
      environment = [
        "HOME=/ephemeral_build_home",
        "TMPDIR=/ephemeral_tmp_dir"
      ]
      [runners.kubernetes]
        [[runners.kubernetes.volumes.empty_dir]]
          name = "build-home"
          mount_path = "/ephemeral_build_home"
          # medium = "Memory"
        [[runners.kubernetes.volumes.empty_dir]]
          name = "tmp-dir"
          mount_path = "/ephemeral_tmp_dir"
          # medium = "Memory"
